### PR TITLE
Kubernetes support

### PR DESCRIPTION
**Pull request by anthonydahanne**

### CODE REVIEW

File: fullstack/README.md [hunk 1]
```diff
@@ -225,3 +225,11 @@ Start several webapp instances :
...
     
     docker service ls
+
+## Kubernetes deployment
+
+With a correctly sized Kubernetes cluster :
+
+    kubectl apply -f src/main/kubernetes
+    
+And oepn the app, play around, start scaling, etc.
```
Comment (henri-tremblay) [inline fullstack/README.md]: typo

File: fullstack/src/main/kubernetes/fullstack-deployment.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: fullstack
+    spec:
+      containers:
+      - name: fullstack
+        image: anthonydahanne/ehcache3-fullstack-demo:latest
```
Comment (henri-tremblay) [inline fullstack/src/main/kubernetes/fullstack-deployment.yml]: I think we should reference the local image built with maven

File: fullstack/src/main/kubernetes/fullstack-deployment.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        imagePullPolicy: Always
+        env:
+        - name: SPRING_PROFILES_ACTIVE
+          value: prod, k8s
+        - name: SPRING_DATASOURCE_URL
+          value: jdbc:mysql://fullstack-mysql:3306/demo?useUnicode=true&characterEncoding=utf8&useSSL=false
+        - name: JAVA_OPTS
+          value: "-Xmx1024m -Xms256m"
```
Comment (henri-tremblay) [inline fullstack/src/main/kubernetes/fullstack-deployment.yml]: Weird to have them in reverse order

File: fullstack/src/main/kubernetes/fullstack-deployment.yml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        - name: JHIPSTER_SLEEP
+          value: "10"
+        - name: APPLICATION_CLUSTER_URI
+          value: "terracotta://demo-terracotta-server1:9410,demo-terracotta-server2:9410/demo-entity"
+        - name: APPLICATION_STUBWEBSERVICES
+          value: "false"
+        - name: APPLICATION_GOOGLEAPIKEY
```
Comment (henri-tremblay) [inline fullstack/src/main/kubernetes/fullstack-deployment.yml]: Can't we pass variables?

File: fullstack/src/main/kubernetes/fullstack-mysql.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+      containers:
+      - name: mysql
+        image: mysql:5.7.20
+        env:
+        - name: MYSQL_USER
+          value: "root"
+        - name: MYSQL_ALLOW_EMPTY_PASSWORD
+          value: "yes"
+        - name: MYSQL_DATABASE
+          value: "demo"
```
Comment (henri-tremblay) [inline fullstack/src/main/kubernetes/fullstack-mysql.yml]: variable?

Comment (anthonydahanne) [reply fullstack/src/main/kubernetes/fullstack-mysql.yml]: not for now :-)

**Comment by anthonydahanne** [comment]
@henri-tremblay  please re review / merge
I've successfully deployed this on Azure AKS

**Pull request merged**